yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -469,6 +469,9 @@ private[spark] class ApplicationMaster(
       System.setProperty("spark.submit.pyFiles",
         PythonRunner.formatPaths(args.pyFiles).mkString(","))
     }
+    if (args.primaryRFile != null && args.primaryRFile.endsWith(".R")) {
+      // TODO(davies): add R dependencies here
```

Comment:
Well so this is yes and no - In YARN cluster we construct the correct command line but the launcher script can't find the SparkR location as we right now use SPARK_HOME to find the install location. I'll open a JIRA for this and describe more details.